src/ec/packages/ec-component-expandable/expandable.test.js

```diff
@@ -0,0 +1,18 @@
+import path from 'path';
+import { renderTwigFileAsNode } from '@ecl-twig/test-utils';
+
+import demoData from './demo/data';
+
+// Add SVG icon path.
+demoData.button.icon.path = 'example';
+
+describe('EC - Expandable', () => {
```

Comment:
it should be possible to add a snapshot for expanded state, to be sure that markup is correct every time.

Reply:
I am two hands for having a test for expanded state! I tried the following so far:

1) I tried to `then` the `expect(render(demoData))` of the test (`await` yields declaration error) using similar approach from the way we include [DOM-related JS](https://github.com/ec-europa/ecl-twig/blob/master/src/ec/packages/ec-component-breadcrumb/breadcrumb.story.js#L57) in stories, and `var button = document.querySelector("[data-ecl-expandable-toggle=true]");` with `button.click` which would trigger the DOM state change, but `jest` does not do anything, of course, the script will go in the snapshot as a DOM contents, but won't run. For this to work, we need a headless browser, I guess.

2) I tried to approach the issue from using a `boolean` flag exposed to the public. Thus, aiming fro a "knob" and a way to immitate state by checks in twig template. It won't be pretty, and we'll need to discuss new properties. For instance, do we expose only 1 property `aria-expanded` which modifies the rest down the road, or do we expose properties separately?

We have the following elements to take into account in order to render expanded state without DOM manipulation, but by input data to the template:

- `aria-expanded` on button. This brings the question: is this flag enough as a public "API"
- button label inside `ecl-button__label` span
- `hidden` attribute on `ecl-expandable__content`. This brings the question: wouldn't it better to expose "extra_attributes" type of variable for content-related attributes?

Reply:
I think that your second option is valid (at least that's probably what I would have do).
If we expose a parameter `default_state` (or something like that), which will control the initial `aria-expanded` and `hidden` it should be good

Reply:
I made a few updates with the suggested approach. I'm afraid, however, the JS behaviors of changing labels is not in the snapshot.